metsim/metsim.py

```diff
@@ -55,87 +81,101 @@ def run(self, method, forcings):
+        if not os.path.exists(out_dir):
+            os.makedirs(out_dir)
+
+        out_file = os.path.join(out_dir, "forcing.nc")
         self.output = Dataset(out_file, 'w')
 
-        # Number of timesteps 
-        n_ts = len(pd.date_range(self.params['start'], 
-                self.params['stop'] + pd.Timedelta('1 days'), 
-                freq=self.params['time_step']+'T'))
-        
+        # Number of timesteps
+        n_ts = len(pd.date_range(MetSim.params['start'],
+                   MetSim.params['stop'] + pd.Timedelta('1 days'),
+                   freq="{}T".format(MetSim.params['time_step'])))
+
         # Create dimensions
         self.output.createDimension('lat', len(self.lats))
         self.output.createDimension('lon', len(self.lons))
-        self.output.createDimension('time', n_ts) 
+        self.output.createDimension('time', n_ts)
 
         # Create output variables
-        for varname in self.params['out_vars']:
+        for varname in MetSim.params['out_vars']:
             self.output.createVariable(varname, 'd', ('time', 'lat','lon'))
 
 
     def write_netcdf(self, data:dict):
-        """Write out to the output file"""
+        """Write out as NetCDF to the output file"""
         print("Writing netcdf...")
         lats = data.keys()
         for lat in lats:
-            lons = data[lat].keys()
-            for lon in lons:
-                for varname in self.params['out_vars']:
-                    self.output.variables[varname][:, lat, lon] = data[lat][lon][varname].values
+            for lon in data[lat].keys():
+                for varname in MetSim.params['out_vars']:
+                    self.output.variables[varname][:, lat, lon] = (
+                            data[lat][lon][varname].values)
 
 
     def write_ascii(self, data: dict):
-        """ TODO """
+        """Write out as ASCII to the output file"""
         print("Writing ascii...")
-        raise NotImplementedError 
-   
+        out_dir = MetSim.params.get('out_dir', './results/')
+        if not os.path.exists(out_dir):
+            os.makedirs(out_dir)
+        lats = data.keys()
+        for lat in lats:
+            for lon in data[lat].keys():
+                out_file = '_'.join(["forcing", str(lat), str(lon)])
+                data[lat][lon].to_csv(os.path.join(out_dir, out_file), sep='\t')
+
 
     def read_binary(self, fpath: str) -> xr.Dataset:
         """ Reads a binary forcing file (VIC 4 format) """
-        dates = pd.date_range(self.params['start'], self.params['stop'])
+        dates = pd.date_range(MetSim.params['start'], MetSim.params['stop'])
         n_days = len(dates)
         precip = [] # Short unsigned int
         t_max  = [] # Short int
         t_min  = [] # Short int
         wind   = [] # Short int
-        
+
         # Pack these for nicer syntax in the loop
         var_name = [precip, t_max, t_min, wind]
         scale = [40.0, 100.0, 100.0, 100.0]
```

Comment:
scale values should be read from the configuration file (or the netcdf file in case of netcdf)